### ISSUE TITLE
Home and End should be supported by keyboard navigation

### ISSUE THREAD

**Issue by CharlesStover** [schrodinger/fixed-data-table-2#319]
Pressing up, down, page up, page down, home, and end while having the fixed-data-table-2 focused should scroll the table up and down the same way the mouse wheel does.

I've had to implement this by having the parent component update with a new scrollTop, but there are drawbacks to this -- namely the re-render time is noticeably laggy compared to the integrated mouse wheel listener. Further, determining the max scroll top is a complicated issue in itself.

Any chance this could get added?

**Comment by wcjordan** [comment]
Check out keyboardScrollEnabled & keyboardPageEnabled.  I think they may do what you ask, but were added relatively recently so I'd consider them in beta.  Let me know if that covers this or if there's more.

API docs can be found here: http://schrodinger.github.io/fixed-data-table-2/api-table.html

**Comment by CharlesStover** [comment]
Works great. I'd suggest adding support for HOME and END (namely END being the hardest to implement by a third party).

Thanks!

**Comment by CharlesStover** [comment]
>for what I suspect is the same functionality

Are you saying Page Up and Page Down are the same functionality as Home and End?
Home would set scrollTop to 0. End would set scrollTop to its maximum possible value.

Page Up and Page Down scroll 100% height.

So, in a table that's 300px tall and each row is 10px with 90 rows, Page Down would set the scrollTop to 300 (30 visible rows * 10px each), while End would set the scrollTop to 600 (90 total rows * 10px each minus 30  visible rows * 10px each).

As different keys, they shouldn't conflict.

I'd dig into a PR, but work deadlines are tight at the moment.

**Comment by wcjordan** [comment]
Ah ok, I misunderstood the desired behavior.  In that case I think you could add the functionality in a similar way to that commit and just use a new separate API flag for it.
